apps/meteor/server/models/raw/Roles.ts

```diff
@@ -48,6 +48,42 @@ export class RolesRaw extends BaseRaw<IRole> implements IRolesModel {
 		return true;
 	}
 
+	async addRolesByUserIds(userIds: IUser['_id'][], roles: IRole['_id'][]): Promise<boolean> {
+		if (!Array.isArray(userIds)) {
+			userIds = [userIds];
+			console.warn('[WARN] Roles.addRolesByUserIds: userIds should be an array');
+		}
+
+		if (!Array.isArray(roles)) {
+			roles = [roles];
+			console.warn('[WARN] Roles.addRolesByUserIds: roles should be an array');
+		}
+
+		const users = await Users.find({ _id: { $in: userIds } }, { projection: { _id: 1 } }).toArray();
+
+		const userIdsToAddRoles = users.map((user) => user._id);
+
+		for await (const roleId of roles) {
```

Comment:
roles here isn't an async iterartor

Reply:
Yes, it is not, but the `await` keyword is necessary in this case to ensure that the loop waits for each asynchronous operation to complete before moving on to the next iteration.

Reply:
But you are awaiting each promise inside the loop anyway. How does this benefit us, while it'll make the process slower because of repeated unwrapping.

Reply:
It won't work without it, I already tried. :)